src/scrcmd.c

```diff
@@ -2377,6 +2377,7 @@ bool8 ScrCmd_updatecoinsbox(struct ScriptContext *ctx)
 
 bool8 ScrCmd_trainerbattle(struct ScriptContext *ctx)
 {
+    TrainerBattleLoadArgs(ctx->scriptPtr);
     Script_RequestEffects(SCREFF_V1 | SCREFF_TRAINERBATTLE);
```

Comment:
```suggestion
    TrainerBattleLoadArgs(ctx->scriptPtr);

    Script_RequestEffects(SCREFF_V1 | SCREFF_TRAINERBATTLE);
```
Minor style suggestion, separating parsing of arguments from the `Script_RequestEffects` call.

I'm pretty sure `TrainerBattleLoadArgs` doesn't have any effect other than to set up variables that aren't in the save, but if it did happen to alter hardware registers or the save it would have to come after `Script_RequestEffects` and you'd need to include `SCREFF_HARDWARE` or `SCREFF_SAVE` in the parameters.

Reply:
`TrainerBattleLoadArgs` should only write to ram. I feel like it should go after `Script_RequestEffects` anyway though